### PR TITLE
Initial changes for v1beta2 CRD

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1047
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/895
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/833
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/551

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Discussion

This adds a new CRD v1beta2, which will be the default stored version. Internally the operator will use v1beta1 until we change this in the next release.

# Testing

Unit tests + local tests.

I tested different settings:

- Cluster v1beta1 -> v1beta2 (worked if old/deprecated settings are there they will be dropped)
- Creating new clusters

# Documentation

Added some additional docs.

# Follow-up

After this we can change the default internal type of the operator to v1beta2 and remove some additional code.

### CODE REVIEW

**Comment by johscheuer** [comment]
The e2e test pipeline will fail until we upgrade the CRD in our test cluster(s).

**Review by brownleej — APPROVED** [review]
👍

File: config/samples/apps_v1beta2_foundationdbcluster.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: apps.foundationdb.org/v1beta2
+kind: FoundationDBCluster
+metadata:
+  name: foundationdbcluster-sample
+spec:
+  # TODO(user): Add fields here
```
[the remark this answers is not in the document]
Comment (johscheuer) [reply config/samples/apps_v1beta2_foundationdbcluster.yaml:6]: I'm going to clean them up later.

**Review by johscheuer — COMMENTED** [review]
I'm going to rebase those changes and add all changes that were merged for v1beta1 to v1beta2 after that I would merge those changes and start writing the change-log for `1.0.0`

**Comment by johscheuer** [comment]
I think I have all deprecations updated for v1beta2.

**Pull request merged**